lib/backend/block_affinity.go

```diff
@@ -44,7 +44,19 @@ func (key BlockAffinityKey) asEtcdKey() (string, error) {
 }
 
 func (key BlockAffinityKey) asEtcdDeleteKey() (string, error) {
-	return key.asEtcdKey()
+	// Must specify at least a host to remove.
+	if key.Host == "" {
+		return "", common.ErrorInsufficientIdentifiers{}
+	}
+	k := "/calico/ipam/v2/host/" + key.Host
```

Comment:
Unsure about this implementation.  It allows you to delete the entire host tree for a given host if you don't specify a CIDR in the key.

In a way it is nice - you're essentially deleting all of the host affinities for a given host and thus it removes the entire host tree for that host, because affinities are all that's stored in that tree.

At the same time, if we ever want to store additional data under `/calico/ipam/v2/host`, then this would end up removing more than just the affinities and we'd need to change the behavior.

Alternative to this is to make a backend IPAMHost resource.